File: se.cpp
#include <opencv2/opencv.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/tracking.hpp>
#include <opencv2/core/ocl.hpp>
#include <stdio.h>
#include <sys/socket.h>
#include <string.h>
#include <arpa/inet.h>
#include <unistd.h>
#include <netinet/in.h>
#include <time.h>
using namespace cv;

void camera_fun(char ip[],int port ,int device,int quality){

	struct sockaddr_in addrRec;

	addrRec.sin_family = AF_INET;
	addrRec.sin_port = htons(port);//
	addrRec.sin_addr.s_addr = inet_addr(ip);//
	
	int sockClient = socket(AF_INET, SOCK_DGRAM, 0);

	if(sockClient < 0){
		printf("%d create socket fail\n",port);
		return ;
	}

	int nLen = sizeof(sockaddr_in);
 
	VideoCapture capture(device);
	Mat capframe;
	
	std::vector<int> params;
	params.push_back(IMWRITE_JPEG_QUALITY);
	params.push_back(quality);

	while (1)
	{
		capture >> capframe;
		std::vector<uchar> buff;
		imencode(".jpg", capframe, buff, params);
	
		Mat frame = imdecode(buff, -1);
		sendto(sockClient, (char *)&buff[0],buff.size() , 0, (struct sockaddr*)&addrRec, nLen);
		waitKey(10);
	}


	close(sockClient);
}
int main(int argc,char **argv)
{
	char *ip=argv[1];
	int port = atoi(argv[2]);
	int device = atoi(argv[3]);
	int quality = atoi(argv[4]);
	 
	camera_fun(ip,port,device,quality);

    return 0;
}

File: se.py
import cv2
import numpy as np
import socket
import sys 
#127.0.0.1 5090 0 50
skt  = socket.socket(socket.AF_INET,socket.SOCK_DGRAM)

cap=cv2.VideoCapture(int( sys.argv[3]) )

while True:
	ret,frame=cap.read()

	img_encode = cv2.imencode('.jpg', frame,[int(cv2.IMWRITE_JPEG_QUALITY), int( sys.argv[4] )] )[1]

	str_encode = np.array(img_encode).tostring()
	nparr = np.fromstring(str_encode, np.uint8)

	skt.sendto(img_encode,(sys.argv[1],int( sys.argv[2] ) ) )

	cv2.waitKey(1)


cap.release()
cv2.destroyAllWindows()
